chainstate/tx-verifier/src/transaction_verifier/token_issuance_cache.rs

```diff
@@ -69,23 +72,68 @@ impl TokenIssuanceCache {
 
     // Token registration saves the token id in the database with the transaction that issued it, and possibly some additional auxiliary data;
     // This helps in finding the relevant information of the token at any time in the future.
-    pub fn register(
+    pub fn register<E>(
         &mut self,
+        chain_config: &ChainConfig,
         block_id: Option<Id<Block>>,
         tx: &Transaction,
-    ) -> Result<(), TokensError> {
-        let was_token_issued = get_tokens_issuance_count(tx.outputs()) > 0;
+        tx_source: &TransactionSourceForConnect,
+        token_data_getter: impl Fn(&TokenId) -> Result<Option<TokenAuxiliaryData>, E>,
+    ) -> Result<(), ConnectTransactionError>
+    where
+        ConnectTransactionError: From<E>,
+    {
+        let was_token_issued = get_tokens_issuance_v0_count(tx.outputs()) > 0;
 
         if was_token_issued {
+            // Check if v0 tokens are allowed to be issued at this height
```

Comment:
Open question is what to do with v0 tokens. Right now I forbid issuance of new v0 after the netupgrade. But transfers of v0 are still allowed. Is it ok?
Also we could upgrade v0 to v1 by spending it but I'm not sure if it's worth implementing.

Part of me still wants to hardfork to avoid the legacy of OutputValue::TokenV0 and the outdated issuances which will haunt us in the codebase forever.

Reply:
Forbidding issuance is a good start. I would also forbid using them completely to be honest as I don't really care. But that should strictly happen after the fork. On mainnet, we'll completely delete that code.

Reply:
as discussed all v0 operations are now forbidden after the v1 fork